### DISCUSSION TITLE
Upserting One to Many Relationships

### DISCUSSION THREAD

**Discussion by colleen-love** [prisma/prisma#13043] [Q&A]
I dug through the generated code, the docs and some previous issues but couldn't find anything that answered this question. 

For the following schema:

```
model User {
  id              String @id @default(uuid())
  email           String @unique()
  permissions Permission[]
}
model Permission {
  id           String @id @default(uuid())
  user         User @relation(fields: [userId], references: [id])
  userId       String   
}
```

Is it possible to upsert the Permission data with `prisma.user.upsert({...})` in a way which does not create duplicate Permission rows _**without making another query to retrieve the user ID**_? This question also applies to updating, but I thought upserting was more relevant as there is a chance that the user doesn't even exist yet.

**Comment by austincrim** [comment]
Hey @collinscangarella 👋 ,

> without making another query to retrieve the user ID

Does this mean that the *generated* Prisma SQL doesn't query for a user's id or that you don't want to manually query for a user's id?

If it's the latter, you can do a nested upsert:
```ts
let result = await prisma.user.upsert({
  where: { id: someId },
  create: { email: 'austin@gmail.com' },
  update: {
    permissions: {
      upsert: {
        where: { id: 'foo' },
        create: { id: 'foo' },
        update: { id: 'bar' }
      }
    }
  }
})
```

**Reply by colleen-love** [reply]
Gotcha. There's no way of knowing the id of the permissions associated with a user without knowing the id of the user.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
